cmasher/utils.py

```diff
@@ -233,6 +239,117 @@ def _get_cmap_perceptual_rank(
 
 
 # %% FUNCTIONS
+# This function combines multiple colormaps at given nodes
+def combine_cmaps(
+    cmaps: list[Union[Colormap, str]],
+    *,
+    nodes: Optional[Union[list[float], np.ndarray]] = None,
+    n_rgb_levels: int = 256,
+    combined_cmap_name: str = "combined_cmap",
+) -> LinearSegmentedColormap:
+    """Create a composite matplotlib colormap by combining multiple colormaps.
+
+    Parameters
+    ----------
+        cmaps: list of Colormap or colormap names (str)
+            list of colormaps to combine.
+        nodes: list or numpy array of nodes (float). Defaults: equal divisions.
+            The blending points between colormaps, in the range [0, 1].
+        n_rgb_levels: int. Defaults: 256.
+            Number of RGB levels for each colormap segment.
+        combined_cmap_name: str. Defaults: "combined_cmap".
+            name of the combined Colormap.
+
+    Returns
+    -------
+        Colormap: The composite colormap.
+
+    Raises
+    ------
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note
+    ----
+        The colormaps are combined from low value to high value end.
+
+    References
+    ----------
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
+
+    Examples
+    --------
+        Using predefined colormap names::
+        >>> custom_cmap_1 = combine_cmaps(
+            ["ocean", "prism", "coolwarm"], nodes=[0.2, 0.75]
+        )
+
+        Using Colormap objects::
+        >>> cmap_0 = plt.get_cmap("Blues")
+        >>> cmap_1 = plt.get_cmap("Oranges")
+        >>> cmap_2 = plt.get_cmap("Greens")
+        >>> custom_cmap_2 = combine_cmaps([cmap_0, cmap_1, cmap_2])
+
+    """
+    # Check colormap datatype and convert to list[Colormap]
+    if len(cmaps) == 1:
+        raise ValueError("Cannot combine a single colormap.")
```

Comment:
Let's make sure we know what happens if *no* colormaps is passed too
```suggestion
    if len(cmaps) < 2:
        raise ValueError("Expected at least two colormaps to combine")
```

Reply:
Excellent point here, two birds with one stone.